firebase-common/src/main/java/com/google/firebase/internal/DataCollectionConfigStorage.java

```diff
@@ -43,15 +43,14 @@ public DataCollectionConfigStorage(
       Context applicationContext, String persistenceKey, Publisher publisher) {
     this.applicationContext = directBootSafe(applicationContext);
     this.sharedPreferences =
-        applicationContext.getSharedPreferences(
+        this.applicationContext.getSharedPreferences(
             FIREBASE_APP_PREFS + persistenceKey, Context.MODE_PRIVATE);
     this.publisher = publisher;
     this.dataCollectionDefaultEnabled = new AtomicBoolean(readAutoDataCollectionEnabled());
   }
 
   private static Context directBootSafe(Context applicationContext) {
```

Comment:
You may want to rename this to `deviceProtectedContext` since you changed the logic inside

Reply:
Actually I think this method is even redundant and you can just use `ContextCompat.createDeviceProtectedStorageContext()` directly as it does the same version check.